### PR TITLE
refactor: updated to use newer scc module version

### PR DESCRIPTION
**Pull request by kierramarie**
### Description

The scc module was updated to be able to handle an existing scc instance. The DA has now been updated to use that version of the module instead of handling it itself. It does assume that an scc_workload has already been created and does not create a new one.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: solutions/instances/main.tf [hunk 1]
```diff
@@ -45,7 +45,7 @@ module "kms" {
   providers = {
     ibm = ibm.kms
   }
-  count                       = var.existing_scc_cos_kms_key_crn != null || var.existing_scc_cos_bucket_name != null ? 0 : 1 # no need to create any KMS resources if passing an existing key, or bucket
+  count                       = (var.existing_scc_cos_kms_key_crn != null || var.existing_scc_cos_bucket_name != null) && var.existing_scc_instance_crn == null ? 0 : 1 # no need to create any KMS resources if passing an existing key, or bucket
```
Comment (akocbek) [inline solutions/instances/main.tf:48]: comment  doesn't reflect the implementation code

File: solutions/instances/main.tf [hunk 2]
```diff
@@ -86,7 +86,7 @@ module "cos" {
   providers = {
     ibm = ibm.cos
   }
-  count                    = var.existing_scc_cos_bucket_name == null ? 1 : 0 # no need to call COS module if consumer is passing existing COS bucket
+  count                    = var.existing_scc_cos_bucket_name == null && var.existing_scc_instance_crn == null ? 1 : 0 # no need to call COS module if consumer is passing existing COS bucket
```
Comment (akocbek) [inline solutions/instances/main.tf:89]: comment  doesn't reflect the implementation code

**Review by akocbek — DISMISSED** [review]

File: solutions/instances/main.tf [hunk 3]
```diff
@@ -128,42 +128,26 @@ module "cos" {
...
 
 module "scc" {
-  count                             = var.existing_scc_instance_crn == null ? 1 : 0
   source                            = "terraform-ibm-modules/scc/ibm"
-  version                           = "1.6.3"
+  version                           = "1.8.0"
+  existing_scc_instance_crn         = var.existing_scc_instance_crn
   resource_group_id                 = module.resource_group.resource_group_id
-  region                            = local.scc_instance_region
+  region                            = var.scc_region
```
Comment (akocbek) [inline solutions/instances/main.tf:141]: if existing instance crn is passed, then I believe the region should be parsed from CRN. why this was removed?

File: solutions/instances/main.tf [hunk 4]
```diff
@@ -162,8 +151,8 @@ module "scc" {
   en_instance_crn                   = var.existing_en_crn
   skip_cos_iam_authorization_policy = var.skip_scc_cos_auth_policy
   resource_tags                     = var.scc_instance_tags
-  attach_wp_to_scc_instance         = var.provision_scc_workload_protection
-  wp_instance_crn                   = var.provision_scc_workload_protection ? module.scc_wp[0].crn : null
+  attach_wp_to_scc_instance         = var.provision_scc_workload_protection && var.existing_scc_instance_crn == null
```
Comment (ocofaigh) [inline solutions/instances/main.tf:154]: This means that user cannot attach workload protection instance to an existing SCC instance. @in-1911 I think we talked about this use case. Do we want to support that or not?

File: solutions/instances/provider.tf [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 provider "ibm" {
   ibmcloud_api_key = var.ibmcloud_api_key
-  region           = local.scc_instance_region
+  region           = var.scc_region
```
Comment (ocofaigh) [inline solutions/instances/provider.tf:7]: I think this should still be `local.scc_instance_region` - why would we change it?

**Comment by kierramarie** [comment]
@ocofaigh conflicts have been resolved

**Review by ocofaigh — APPROVED** [review]

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**